cmd/libsnap-confine-private/cgroup-support.c

```diff
@@ -97,3 +98,123 @@ bool sc_cgroup_is_v2() {
     }
     return false;
 }
+
+static bool traverse_looking_for_prefix_in_dir(DIR *root, const char *prefix, const char *skip) {
+    while (true) {
+        errno = 0;
+        struct dirent *ent = readdir(root);
+        if (ent == NULL) {
+            if (errno != 0) {
+                die("cannot read directory entry");
+            }
+            break;
+        }
+        if (ent->d_type != DT_DIR) {
+            continue;
+        }
+        if (sc_streq(ent->d_name, "..") || sc_streq(ent->d_name, ".")) {
+            /* we don't want to go up or process the current directory again */
+            continue;
+        }
+        if (sc_streq(ent->d_name, skip)) {
```

Comment:
Can you please add a comment (either here or where you prepare the leaf string), explaining what we are skipping here and why?

Reply:
i've added a general comment here and a more explanatory one in sc_cgroup_v2_is_tracking_snap()